### PR TITLE
Add WechatJobErrorHandler to support sending message

### LINKED ISSUE #1461: Add WechatJobErrorHandler to support sending message
To support for sending the messages when the job executes with errors, we decide to do some extensions for JobErrorHandler.
- create a new module - elasticjob-error-handler/elasticjob-error-handler-wechat
- load config from System parameters.
- load config from specified conf.
- use http-client pool to send messages.

### PR DESCRIPTION
**Pull request by viviel**
Fixes #1461.

Changes proposed in this pull request:
- create a new module - elasticjob-error-handler/elasticjob-error-handler-wechat.
- load config from specified conf.
- use http-client pool to send messages.

### CODE REVIEW

File: elasticjob-error-handler/elasticjob-error-handler-wechat/pom.xml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            <artifactId>log4j-over-slf4j</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>jcl-over-slf4j</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+</project>
```
Comment (TeslaCN) [inline elasticjob-error-handler/elasticjob-error-handler-wechat/pom.xml:87]: Need a blank line.

File: elasticjob-error-handler/elasticjob-error-handler-wechat/src/main/java/org/apache/shardingsphere/elasticjob/error/handler/wechat/WechatConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    
+    private final Integer readTimeout;
+    
+    /**
+     * Get connect timeout config or default value.
+     *
+     * @return connect timeout value
+     */
+    public Integer getConnectTimeoutOrDefault() {
+        if (Objects.isNull(connectTimeout)) {
```
Comment (TeslaCN) [inline elasticjob-error-handler/elasticjob-error-handler-wechat/src/main/java/org/apache/shardingsphere/elasticjob/error/handler/wechat/WechatConfiguration.java]:
```suggestion
        if (null == connectTimeout) {
```

File: elasticjob-error-handler/elasticjob-error-handler-wechat/src/main/java/org/apache/shardingsphere/elasticjob/error/handler/wechat/WechatConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        return connectTimeout;
+    }
+    
+    /**
+     * Get read timeout config or default value.
+     *
+     * @return read timeout value
+     */
+    public Integer getReadTimeoutOrDefault() {
+        if (Objects.isNull(readTimeout)) {
```
Comment (TeslaCN) [inline elasticjob-error-handler/elasticjob-error-handler-wechat/src/main/java/org/apache/shardingsphere/elasticjob/error/handler/wechat/WechatConfiguration.java]:
```suggestion
        if (null == readTimeout) {
```

**Review by TeslaCN — CHANGES_REQUESTED** [review]

File: elasticjob-error-handler/elasticjob-error-handler-wechat/pom.xml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>elasticjob-error-handler</artifactId>
+        <groupId>org.apache.shardingsphere.elasticjob</groupId>
+        <version>3.0.0-beta-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
```
Comment (Technoboy-) [inline elasticjob-error-handler/elasticjob-error-handler-wechat/pom.xml]: modelVersion should be before parent

File: elasticjob-error-handler/elasticjob-error-handler-wechat/pom.xml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>elasticjob-error-handler</artifactId>
+        <groupId>org.apache.shardingsphere.elasticjob</groupId>
```
Comment (Technoboy-) [inline elasticjob-error-handler/elasticjob-error-handler-wechat/pom.xml:25]: groupId first

**Comment by Technoboy-** [comment]
Miss SPI config

**Comment by TeslaCN** [comment]
Shall we name it 'WeCom'? 
Refer to: https://work.weixin.qq.com/

image

File: elasticjob-error-handler/elasticjob-error-handler-wechat/pom.xml [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+        </dependency>
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>log4j-over-slf4j</artifactId>
+            <scope>test</scope>
```
Comment (Technoboy-) [inline elasticjob-error-handler/elasticjob-error-handler-wechat/pom.xml:78]: No need to specified scope

File: elasticjob-error-handler/elasticjob-error-handler-wechat/pom.xml [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            <artifactId>mockito-core</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
+            <artifactId>logback-classic</artifactId>
+            <scope>test</scope>
```
Comment (Technoboy-) [inline elasticjob-error-handler/elasticjob-error-handler-wechat/pom.xml:73]: No need to specified scope

**Review by Technoboy- — APPROVED** [review]

**Pull request merged**